### PR TITLE
Feat/add auto renewals collections

### PR DESCRIPTION
**Pull request by Th0rgal**
This pull request introduces two collections required to use this mongodb with the auto_renewal_bot. Some collections were common with the main indexer. In order to minimize the cost of maintenance  we will keep a single repository. If possible we will also not deploy an instance dedicated to the auto_renewal_bot.

I updated the collection names so it will require a few changes in the auto_renew_bot.

What you had:
`approvals` -> { renewer, value }
`auto_renewals` -> { domain, renewer_address, limit_price, auto_renewal_enabled, last_renewal, meta_hash }
`domains`-> { domain, expiry, token_id, creation_date, addr, rev_addr }
`domains_renewals` -> { domain, prev_expiry, new_expiry, renewal_date }
`starknet_ids` -> { owner, token_id, creation_date } 

Types used in replacement:
timestamps are u64
felt values are in hex: 0x followed by 64 char bytes
ether values that don't need to be exact are double
ether amounts that need to be exact are a 0x followed by 64 char bytes representing the wei amount

Replacement:
`auto_renew_approvals` ->  { renewer, allowance }
`auto_renew_flows` -> { domain, renewer_address, limit_price, enabled, last_renewal, meta_hash } //  auto_renewal_enabled became enabled
`domains`-> { domain, expiry, id, creation_date, legacy_address, rev_address, root } // token_id became id, *addr became *address and address became legacy_address. root is true if it is a root domain. In the future legacy_address won't be enough to find the target address (if it is null we will have to check the starknetid data)
`renewals` -> { domain, prev_expiry, new_expiry, timestamp } // renewal_date became timestamp
`id_owners` -> { owner, id, creation_date, main } // token_id became id, main is always false for now, will be true if it is the main starknet_id of someone

### CODE REVIEW

File: src/auto_renew_flows.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      fromAddress: formatFelt(AUTO_RENEW_CONTRACT),
+      keys: [formatFelt(SELECTOR_KEYS.ON_AUTO_RENEW_ENABLED)],
+    },
+    {
+      fromAddress: formatFelt(AUTO_RENEW_CONTRACT),
+      keys: [formatFelt(SELECTOR_KEYS.ON_AUTO_RENEW_DISABLED)],
+    },
+    {
+      fromAddress: formatFelt(AUTO_RENEW_CONTRACT),
+      keys: [formatFelt(SELECTOR_KEYS.ON_AUTO_RENEW_ENABLED)],
```
Comment (irisdv) [inline src/auto_renew_flows.ts]: there is twice the event `ON_AUTO_RENEW_ENABLED`, and the case `ON_AUTO_RENEW` is missing in the filters

File: src/auto_renew_flows.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            uint256.uint256ToBN({ low: event.data[2], high: event.data[3] }),
+            DECIMALS
+          );
+          const timestamp = event.data[4];
+
+          return {
+            entity: {
+              domain,
+              renewer_address: renewerAddr,
+              limit_price: +limitPrice,
```
Comment (irisdv) [inline src/auto_renew_flows.ts:119]: why do you use a `+` before `limitPrice` and `timestamp` (below) ?

**Review by irisdv — APPROVED** [review]
Lgtm! Just left a comment and a question in the code

File: src/auto_renew_flows.ts (hunk 2 above)
Comment (Th0rgal) [reply src/auto_renew_flows.ts:119]: this is to tell to mongodb to store them as numbers even though they are not (it will convert it)

Comment (irisdv) [reply src/auto_renew_flows.ts:119]: ah ok thanks !

**Pull request merged**